### PR TITLE
Hts 911

### PR DESCRIPTION
**Pull request by Jackie1988**

### CODE REVIEW

File: app/uk/gov/hmrc/helptosavetestadminfrontend/controllers/DeleteVerifiedEmailsController.scala [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.mvc._
+import uk.gov.hmrc.helptosavetestadminfrontend.config.AppConfig
+import uk.gov.hmrc.helptosavetestadminfrontend.forms.EmailsForm
+import uk.gov.hmrc.helptosavetestadminfrontend.repos.VerifiedEmailMongoRepository
+import uk.gov.hmrc.helptosavetestadminfrontend.views
+
+import scala.concurrent.Future
+
+class DeleteVerifiedEmailsController @Inject()(verifiedEmailRepo: VerifiedEmailMongoRepository)(implicit override val appConfig: AppConfig,
```
Comment (sureshhmrc) [inline app/uk/gov/hmrc/helptosavetestadminfrontend/controllers/DeleteVerifiedEmailsController.scala]: you can rename this as `EmailController` - controller names generally do not specify actions , but resources

File: app/uk/gov/hmrc/helptosavetestadminfrontend/repos/VerifiedEmailMongoRepository.scala [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+import scala.concurrent.{ExecutionContext, Future}
+
+case class Email(email: String)
+
+object Email {
+  implicit val format = Json.format[Email]
+}
+
+@ImplementedBy(classOf[VerifiedEmailMongoRepositoryImpl])
+trait VerifiedEmailMongoRepository {
```
Comment (sureshhmrc) [inline app/uk/gov/hmrc/helptosavetestadminfrontend/repos/VerifiedEmailMongoRepository.scala]: `EmailMongoRepository`

File: app/uk/gov/hmrc/helptosavetestadminfrontend/forms/EmailsForm.scala [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+  def deleteEmailsForm = Form(
+    mapping(
+      "emails" -> nonEmptyText
+    )(Emails.apply)(Emails.unapply)
+  )
+
+}
+
+case class Emails(emails: String)
```
Comment (sureshhmrc) [inline app/uk/gov/hmrc/helptosavetestadminfrontend/forms/EmailsForm.scala]:
```case class Email(email: String)
case class Emails(email: Email)``` 
and can we re-use this `Email` in the mongo store ?

File: app/uk/gov/hmrc/helptosavetestadminfrontend/repos/VerifiedEmailMongoRepository.scala [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    idFormat = ReactiveMongoFormats.objectIdFormats
+  ) with VerifiedEmailMongoRepository {
+
+
+  def deleteEmails(emails: List[String])(implicit ec: ExecutionContext): Future[Either[List[String], Unit]] = {
+
+    val result: List[Future[Either[String, Unit]]] =  emails.map { email ⇒
+      remove("email" -> email).map { res ⇒
+        if (res.hasErrors) {
+          Left(s"An error has occurred while deleting email: $email")
```
Comment (sureshhmrc) [inline app/uk/gov/hmrc/helptosavetestadminfrontend/repos/VerifiedEmailMongoRepository.scala]: you are not returning what the actual error is ?

File: conf/app.routes [hunk 5]
```diff
@@ -1,3 +1,6 @@
 # microservice specific routes
 
-GET        /hello-world             uk.gov.hmrc.helptosavetestadminfrontend.controllers.HelloWorld.helloWorld
+GET        /                            @uk.gov.hmrc.helptosavetestadminfrontend.controllers.DeleteVerifiedEmailsController.specifyEmailsToDelete
+POST       /delete-email                @uk.gov.hmrc.helptosavetestadminfrontend.controllers.DeleteVerifiedEmailsController.deleteVerifiedEmails
```
Comment (sureshhmrc) [inline conf/app.routes]: plural  `/delete-emails` ?

File: conf/application.conf [hunk 6]
```diff
@@ -34,60 +56,63 @@ play.filters.headers.contentSecurityPolicy= "default-src 'self' 'unsafe-inline'
...
-microservice {
-    metrics {
-        graphite {
-            host = localhost
-            port = 2003
-            prefix = play.${appName}.
-            enabled = true
-        }
-    }
+play.crypto.secret="UxhBwXnMmrL4fRKTZwcOduV2nrQhbilfTNZlCqyzR3Vb6VB3gQRjHrZqBd9ZPiMo"
```
Comment (sureshhmrc) [inline conf/application.conf]: is this required ? 
looks like you copied it from some other service ?

File: conf/application.conf [hunk 7]
```diff
@@ -34,60 +56,63 @@ play.filters.headers.contentSecurityPolicy= "default-src 'self' 'unsafe-inline'
...
-      host = localhost
-      port = 8100
+Test {
+
+  services {
+
+  }
+
+  auditing {
+    enabled=true
```
Comment (sureshhmrc) [inline conf/application.conf]: we don't need auditing for this service as its only for non prod env

File: conf/logback.xml [hunk 8]
```diff
@@ -2,21 +2,27 @@
 <configuration>
 
     <appender name="FILE" class="ch.qos.logback.core.FileAppender">
-        <file>logs/help-to-save-test-admin-frontend.log</file>
+        <file>logs/help-to-save-frontend.log</file>
```
Comment (sureshhmrc) [inline conf/logback.xml]: `help-to-save-test-admin-frontend.log`

File: conf/prod.routes [hunk 9]
```diff
@@ -3,4 +3,4 @@
 ->         /                          health.Routes
 ->     	   /template                  template.Routes
 
-GET        /admin/metrics             com.kenshoo.play.metrics.MetricsController.metrics
+GET        /admin/metrics             @com.kenshoo.play.metrics.MetricsController.metrics
```
Comment (sureshhmrc) [inline conf/prod.routes]: do we need metrics ?

File: test/config/FiltersSpec.scala [hunk 10]
```diff
@@ -0,0 +1,76 @@
+/*
```
Comment (sureshhmrc) [inline test/config/FiltersSpec.scala]: personally i don't care about the tests for this non-prod service  -  
if we want tests, then make sure the min coverage is set to atleast `90` not `80` ?

File: app/uk/gov/hmrc/helptosavetestadminfrontend/controllers/DeleteVerifiedEmailsController.scala (hunk 1 above)
Comment (Jackie1988) [reply app/uk/gov/hmrc/helptosavetestadminfrontend/controllers/DeleteVerifiedEmailsController.scala]: Yeah I'm thinking VerifiedEmailController is more suitable

File: app/uk/gov/hmrc/helptosavetestadminfrontend/repos/VerifiedEmailMongoRepository.scala (hunk 2 above)
Comment (Jackie1988) [reply app/uk/gov/hmrc/helptosavetestadminfrontend/repos/VerifiedEmailMongoRepository.scala]: I think VerifiedEmailMongoRepository is better than EmailMongoRepository as the mongo collection is verifiedEmail

File: app/uk/gov/hmrc/helptosavetestadminfrontend/repos/VerifiedEmailMongoRepository.scala (hunk 4 above)
Comment (Jackie1988) [reply app/uk/gov/hmrc/helptosavetestadminfrontend/repos/VerifiedEmailMongoRepository.scala]: good point, I have updated the string to include the errmsg

File: conf/app.routes (hunk 5 above)
Comment (Jackie1988) [reply conf/app.routes]: yes, updated, thanks

File: conf/application.conf (hunk 6 above)
Comment (Jackie1988) [reply conf/application.conf]: No its not, thanks, will remove

File: conf/application.conf (hunk 7 above)
Comment (Jackie1988) [reply conf/application.conf]: ok removed.

File: conf/logback.xml (hunk 8 above)
Comment (Jackie1988) [reply conf/logback.xml]: copy and paste error, sorry lol

File: app/uk/gov/hmrc/helptosavetestadminfrontend/controllers/DeleteVerifiedEmailsController.scala (hunk 1 above)
Comment (sureshhmrc) [reply app/uk/gov/hmrc/helptosavetestadminfrontend/controllers/DeleteVerifiedEmailsController.scala]: ok, thats fine, but the endpoint doesn't say about `verified`

File: app/uk/gov/hmrc/helptosavetestadminfrontend/repos/VerifiedEmailMongoRepository.scala [hunk 11]
```diff
@@ -0,0 +1,79 @@
...
+case class Email(email: String)
+
+object Email {
+  implicit val format = Json.format[Email]
+}
+
+@ImplementedBy(classOf[VerifiedEmailMongoRepositoryImpl])
+trait VerifiedEmailMongoRepository {
+
+  def deleteEmails(emails: List[String])(implicit ec: ExecutionContext): Future[Either[List[String], Unit]]
```
Comment (sureshhmrc) [inline app/uk/gov/hmrc/helptosavetestadminfrontend/repos/VerifiedEmailMongoRepository.scala]: `emails: List[String]` -> `emails: List[Email]`

**Pull request merged**